docs/development/pigs/pig-013.rst

```diff
@@ -0,0 +1,451 @@
+community, and the skills and manpower of the Gammapy developer team. E.g. we
+might start using ``numba`` or ``dask`` or ``ray`` in the near future, or if
+Tensorflow continues to mature and improve in the coming years (see ``jax`` and
+``tensorflow-probability``), that could be considered for Gammapy.
+
+It is not described here, but very likely after implementing this PIG we will
+follow up by improving Gammapy testing (`GH 2270`_) and modernise the Gammapy
+packaging following the recommendations in `APE 17`_, and we might also change
+the Gammapy git and release workflow to have stable branches with backports
+after v1.0.
+
+Alternatives
+============
+
+Since we have a working setup already, we could do nothing, which seems nice at
+first, but really means continued time sink and suffering by the Gammapy
+maintainers of the continuous integration testing system and the release
+manager.
+
+The parts here that require a lot of work are ``astropy-healpix``,
+``uncertainties`` and the binary wheels. If any of those parts gets stuck or we
+don't have the manpower to do it, they can be done later, they aren't blockers
+for v1.0.
+
+For conda, we should strongly consider introducing a ``gammapy-all`` metapackage
+instead of the current ``gammapy-environment.yml`` shipped via ``gammapy.org``.
+That's what e.g. the Fermi tools and glueviz and others do, and they like that
+solution. This could be done any time, that change is independent of the rest of
+the changes described here.
+
+We could support other ways to ship Gammapy, e.g. Homebrew (works on MacOS and
+also Linux now), or Docker images (works anywhere).
+
+Task list
+=========
+
+Following the Gammapy v0.13 release in July 2019, the following changes should
+be made, which will ship to users in Gammapy v0.14 in September 2019:
+
+- Maybe: update Gammapy packaging and dev tools to follow new recommendations in `APE 17`_.
+  There's some advantages to doing this first, to make it easier to define the
+  new CI test matrix, but this could also come later.
+- Update continuous integration test matrix to test against the minimum required
+  versions as specified here (and also newer versions in addition) (`GH 2270`_).
+  Maybe move completely from travis-ci to Azure?
+- Drop Python 3.5 support and modernise codebase (e.g. use f-strings and use
+  dict instead of OrderedDict).
+- Review Gammapy codebase and remove workarounds for old Python, Numpy, Astropy,
+  ... versions.
+- Update ``extras_require`` in ``setup.py``. Try moving the spec out of
+  ``setup.py`` to ``setup.cfg`` or ``pyproject.toml``.
+- Review, restructure and update all Gammapy installation instructions
+- Change ``astropy-healpix`` to the CDS implementation, release a stable version and
+  check conda and pip binaries (`astropy-healpix GH 128`_)
+- Change Gammapy from ``healpy`` to ``astropy-healpix`` (`GH 1167`_)
+- Maybe: rewrite spectral model error evaluation code to not use the
```

Comment:
Maybe leave that for PIG14

Reply:
OK, removed that point here. (although there's no problem in having some overlap in PIGs, the important thing is to get good solutions, agree on them, and implement them, quickly)